cookbook/controller/service.rst

```diff
@@ -232,6 +234,87 @@ inject *only* the exact service(s) that you need directly into the controller.
    are valid, exactly how you want to organize your reusable code is up to
    you.
 
+Base Controller Methods and their Service Replacements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This list explains how to replace the convenience methods of the base
+controller:
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::createForm` (service: ``form.factory``)
+    .. code-block:: php
+
+        $formFactory->create($type, $data, $options);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::createFormBuilder` (service: ``form.factory``)
+    .. code-block:: php
+
+        $formFactory->createBuilder('form', $data, $options);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::createNotFoundException`
+    .. code-block:: php
+
+        throw new NotFoundHttpException($message, $previous);
```

Comment:
This is not exactly the same. The `Controller` class doesn't throw the exception.